x-pack/plugins/security_solution/server/lib/detection_engine/signals/filters/filter_events.ts

```diff
@@ -21,13 +20,17 @@ export const filterEvents = <T>({
   return events.filter((item) => {
     return fieldAndSetTuples
       .map((tuple) => {
-        const eventItem = get(tuple.field, item._source);
-        if (eventItem == null) {
-          return true;
-        } else if (tuple.operator === 'included') {
+        const eventItem = item.fields ? item.fields[tuple.field] : undefined;
+        if (tuple.operator === 'included') {
+          if (eventItem == null) {
+            return true;
+          }
           // only create a signal if the event is not in the value list
           return !tuple.matchedSet.has(JSON.stringify(eventItem));
         } else if (tuple.operator === 'excluded') {
+          if (eventItem == null) {
+            return false;
+          }
           // only create a signal if the event is in the value list
           return tuple.matchedSet.has(JSON.stringify(eventItem));
         } else {
```

Comment:
Since there's only two operators, not sure this else path ever hits.

Reply:
It shouldn't be hit, but since the operator is a string it's possible for it to get into an invalid state so it's good to handle that possibility. Looking at it again we'd probably want to return `true` in that case so an invalid exception operator doesn't allowlist everything, but I think that's an issue for a separate PR.

Reply:
👍  Wonder if we'd want to log there too. Worry about just letting through invalid states.